src/generic_ui/polymer/cloud-install.ts

```diff
@@ -39,7 +39,12 @@ Polymer({
     this.$.loginOverlay.open();
   },
   launchDigitalOceanSignup: function() {
-    ui.openTab('https://cloud.digitalocean.com/registrations/new?refcode=5ddb4219b716');
+    // DigitalOcean referral codes trump promo codes,
+    // so only add our refcode to the url if the user has no promo code.
+    const hasPromo = ui_context.model.globalSettings.promo === 'off2016'; // TODO: move to constant
+    const urlHasPromo = 'https://cloud.digitalocean.com/registrations/new';
```

Comment:
While this happens to be the url we want to use if the user has a promo code, I think that the more important thing about it is it is the base url (and the line below this is the base url with an additional refcode added).

I think it would be more clear as
```
const registerUrl = '...';
const registerUrlWithReferral = registerURl + '...';
```